SuggestedNameAndDir.md

```diff
@@ -0,0 +1,240 @@
+suggested file name with one that is accepted). This seems less desirable than 3
+because this would mean that specifying an extension in the `suggestedName` is
+optional as long as `excludeAcceptAllOption` is set to true, but then later
+changing `excludeAcceptAllOption` to false would suddenly change behavior and
+possibly break existing API usage.
+
+### Interaction between `startIn`, `startInParentOf` and `id`
+
+All these attributes influence what directory the file picker should start with,
+as such it isn't immediately obvious what should happen if all are provided.
+
+Our proposal is for it to be an error to provide both `startIn` and
+`startInParentOf`. It should not be an error to provide both `startIn*` and
+`id`. If both are provided, `startIn*` will specify what directory the file
+picker should start out it, and the ultimately picked directory will be recorded
+as the last selected directory for the given `id`, such that future invocations
+with that `id` but no `startIn*` value will use the directory.
+
+#### Considered alternatives
+
+We could reject if both a `startIn` option and `id` are provided, to ensure only
+one option controls which directory to start in. There don't see to be many
+downsides to allowing both though, as recording the last used directory for
+future invocations of a file picker without a `startIn` option still seems
+useful.
+
+We could also not reject if both `startIn` and `startInParentOf` are provided.
+This might have benefits if browsers only implement one of them. Websites
+could still get some of the behavior they want without having to do tricky
+feature detection to figure out which feature is and isn't supported. On the
+other hand, allowing both would mean we'd have to specify a ordering between
+them, and unless the ordering we specify happens to match the fallback behavior
+a website wants, websites would still need to feature detect. Only allowing one
+is simpler to reason about.
+
+We could allow file handles to be passed to `startIn` as well. This could behave
+as if the file handle was passed to `startInParentOf`, while the name of the
+file was passed as `suggestedName`. This would perhaps make the Save As use
+cases somewhat simpler, but is also redundant with what is already possible with
+the proposed API. Furthermore we would then have to specify what happens if both
+a `startIn` file, and an explicit `suggestedName` is specified.
+
+Finally, rather than having both `startIn` and `startInParentOf`, we could have
+a single option that accepts both a file or a directory handle. If a file is
+passed in, it behaves as the above described `startInParentOf`, but if a
+directory is passed in it behaves as `startIn`. This means we would lose the
+ability to start a picker in the parent directory of a given directory handle,
+but that is also one of the more edge use cases, and only having a single option
+to deal with otherwise simplifies the API somewhat.
```

Comment:
Having just a flexible `startIn` sounds a little more straightforward than having both. Another alternative could be to make file handles aware of their parent folder, so you could do `startIn: fileHandle.getParentFolder()`.

Reply:
I changed it to have just a single `startIn` as my preferred option. I agree that it seems more straightforward.

Adding `handle.getParent()` or something like that comes up every now and then. We so far have resisted doing so, because it is actually not always very clear what such a method would return. Especially how it would interact with permissions is non obvious (would it only return something if you happen to already have read access to it somehow? In which case it wouldn't be very useful for this use case. But otherwise we have to be very careful not to accidentally reveal too much about the structure of directories on your disk). In general things seems much easier to reason about if "get the parent" isn't an operation that is provided by the API. If you want getParent() to just return "the directory handle (if any) this file/directory was received via getFileHandle/getDirectoryHandle" the website can just keep track of that themselves.